tests/common/dualtor/mux_simulator_control.py

```diff
@@ -107,114 +115,145 @@ def _post(server_url, data):
+        Args:
+            interface_name: a str, the name of interface
+            directions: a list, may contain "upper_tor", "lower_tor", "nic"
+        Returns:
+            None.
+        """
+        server_url = url(interface_name, DROP)
+        data = {"out_ports": directions}
+        pytest_assert(_post(server_url, data), "Failed to set drop on {}".format(directions))
 
-def set_output(mux_server_url, physical_port, directions):
-    """
-    Function to set output for a certain direction on a port
-    Args:
-        mux_server_url: a str, the address of mux server
-        physical_port: physical port on switch, an integer starting from 1
-        directions: a list, may contain "upper_tor", "lower_tor", "nic"
-    Returns:
-        None.
-    """
-    server_url = _url(mux_server_url, physical_port, OUTPUT)
-    data = {"out_ports": directions}
-    pytest_assert(_post(server_url, data), "Failed to set output on {}".format(directions))
+    return _set_drop
 
-def _simulator_port_toggle_to(mux_server_url, physical_port, target):
-    """
-    A helper function to toggle y_cable simulator ports
-    Args:
-        mux_server_url: a str, the address of mux server
-        physical_port: physical port on switch, an integer starting from 1
-        target: "upper_tor" or "lower_tor"
-    """
-    server_url = _url(mux_server_url, physical_port)
-    data = {"active_side": target}
-    pytest_assert(_post(server_url, data), "Failed to toggle to {} on port {}".format(target, physical_port))
+@pytest.fixture
+def set_output(url):
+    """
+    A helper function is returned to make fixture accept arguments
+    """
+    def _set_output(interface_name, directions):
+        """
+        Function to set output for a certain direction on a port
+        Args:
+            interface_name: a str, the name of interface
+            directions: a list, may contain "upper_tor", "lower_tor", "nic"
+        Returns:
+            None.
+        """
+        server_url = url(interface_name, OUTPUT)
+        data = {"out_ports": directions}
+        pytest_assert(_post(server_url, data), "Failed to set output on {}".format(directions))
 
-def toggle_simulator_port_to_upper_tor(mux_server_url, physical_port):
-    """
-    Function to toggle a given y_cable ports to upper_tor
-    Args:
-        mux_server_url: a str, the address of mux server
-        physical_port: physical port on switch, an integer starting from 1
-    """
-    _simulator_port_toggle_to(mux_server_url, physical_port, UPPER_TOR)
+    return _set_output
 
-def toggle_simulator_port_to_lower_tor(mux_server_url, physical_port):
-    """
-    Function to toggle a given y_cable ports to lower_tor
-    Args:
-        mux_server_url: a str, the address of mux server
-        physical_port: physical port on switch, an integer starting from 1
-    """
-    _simulator_port_toggle_to(mux_server_url, physical_port, LOWER_TOR)
+@pytest.fixture
+def toggle_simulator_port_to_upper_tor(url):
```

Comment:
Is there any specific reason why there are diff fixtures for toggle to upper and lower? Can these be combined to toggle_simulator_port_to_tor(url, target_tor)

Reply:
Yes. It's possible to merge ```toggle_simulator_port_to_upper_tor``` and ```toggle_simulator_port_to_lower_tor``` into a common fixture ```toggle_simulator_port_to_tor```, but that's means the caller have to know more internal details and pass correct arguments to the interface, such as ```lower_tor``` or ```upper_tor```. Spliting them into two separated call would be more friendly to caller, I think.